cmd/api/src/database/graphschema.go

```diff
@@ -94,6 +95,111 @@ func (s *BloodhoundDB) GetGraphSchemaExtensionById(ctx context.Context, extensio
 	return extension, nil
 }
 
+type DBFilter struct {
+	Column   string
+	Operator string
+	Value    any
+	IsOr     bool
+}
+
+type DBSort struct {
+	Column        string
+	SortAscending bool
+}
+
+func DBFilterToSql(filters []DBFilter) (string, []any) {
+	var (
+		filterSqlStr    string
+		filterSqlParams []any
+	)
+	if len(filters) > 0 {
+		filterSqlStr += " WHERE "
+		for i, filt := range filters {
+			filterSqlParams = append(filterSqlParams, filt.Value)
+			filterSqlStr += filt.Column + " " + filt.Operator + " ? "
+			if i < (len(filters) - 1) {
+				if filt.IsOr {
+					filterSqlStr += "OR "
+				} else {
+					filterSqlStr += "AND "
+				}
+			}
+		}
+	}
+	return filterSqlStr, filterSqlParams
+}
+
+func DBSortToSql(sorts []DBSort) string {
+	var (
+		sortColumns []string
+		orderSqlStr string
+	)
+	if len(sorts) > 0 {
+		for _, item := range sorts {
+			dirString := "ASC"
+			if !item.SortAscending {
+				dirString = "DESC"
+			}
+			sortColumns = append(sortColumns, item.Column+" "+dirString)
+		}
+		orderSqlStr = "ORDER BY " + strings.Join(sortColumns, ", ")
+	}
+	return orderSqlStr
+}
```

Comment:
The helpers should not be public right now. Moving them to `helper.go` is probably fine for now.